File: app_frontend/src/types/bono.ts
// src/types/bono.ts

export interface BonoCalculationRequest {
  start_date: string;
  end_date: string;
  vendedor_id?: number | null;
}

export interface BonoFacturaDetalle {
    factura_id: number;
    numero_orden: string | null;
    // --- NUEVO CAMPO AÑADIDO ---
    razon_social_cliente: string; 
    honorarios: number;
    gastos: number;
    neto: number;
    porcentaje_aplicado: number;
    bono_generado: number;
}

export interface BonoVendedorResult {
  vendedor_id: number;
  nombre_vendedor: string;
  rut_vendedor: string;
  total_honorarios: number;
  total_gastos: number;
  total_neto: number;
  bono_calculado: number;
  detalle_facturas: BonoFacturaDetalle[];
  porcentaje_bono_aplicado?: number;
}

export interface BonoCalculationResponse {
  start_date: string;
  end_date: string;
  resultados: BonoVendedorResult[];
}

File: app_backend/app/schemas/bono.py
# app/schemas/bono.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import date

# Schema para la solicitud de cálculo
class BonoCalculationRequest(BaseModel):
    start_date: date
    end_date: date
    vendedor_id: Optional[int] = Field(None, description="ID del vendedor para calcular. Si es None, se calculan todos.")

# Schema para el resultado de un vendedor
class BonoVendedorResult(BaseModel):
    vendedor_id: int
    nombre_vendedor: str
    rut_vendedor: str
    total_honorarios: float
    total_gastos: float
    total_neto: float
    bono_calculado: float
    detalle_facturas: List[dict] # Un diccionario simple para el detalle

# Schema para la respuesta completa de la API
class BonoCalculationResponse(BaseModel):
    start_date: date
    end_date: date
    resultados: List[BonoVendedorResult]